hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java

```diff
@@ -521,7 +535,13 @@ protected List<IBaseResource> toResourceList(ISearchBuilder theSearchBuilder, Li
 		List<IBaseResource> resources = new ArrayList<>();
 		theSearchBuilder.loadResourcesByPid(thePids, includedPidList, resources, false, myRequest);
 
+		// we will send the resource list to our interceptors
+		// this can (potentially) change the results being returned.
+		int precount = resources.size();
 		resources = ServerInterceptorUtil.fireStoragePreshowResource(resources, myRequest, myInterceptorBroadcaster);
+		theResponsePageBuilder.setOmittedResourceCount(precount - resources.size());
```

Comment:
question: can myOmittedResource count be non-zero before this?  If so, then this should be 
```suggestion
		theResponsePageBuilder.setOmittedResourceCount(theResponsePageBuilder.getOmittedResourceCount() + precount - resources.size());
```

Reply:
probably not. But i've updated the code to allow for this case anyways.